jlm/llvm/opt/reduction.cpp

```diff
@@ -111,11 +113,43 @@ enable_binary_reductions(rvsdg::Graph & graph)
   nf->set_mutable(true);
   nf->set_reducible(true);
 }
+#endif
+
+static bool
+NormalizeLoadNode(LoadNonVolatileNode & node)
+{
+  static std::vector<rvsdg::NodeNormalization<LoadNonVolatileOperation>> loadNormalizations(
+      { NormalizeLoadMux, NormalizeLoadAlloca });
+
+  auto normalizeLoadNode = std::bind(
+      rvsdg::NormalizeSequence<LoadNonVolatileOperation>,
+      loadNormalizations,
+      std::placeholders::_1,
+      std::placeholders::_2);
```

Comment:
I did not know that `std::placeholders::*` was a thing, but they are kind of cool.

One thing to note is that this call to `bind` takes `loadNormalizations` by value, which means it is copied when constructing `normalizeLoadNode`.

In order to make the bind object not copy our vector, we can use `std::cref(loadNormalizations)`.

Reply:
Reading the documentation of `std::bind` was difficult, so I tested in godbolt to confirm: https://godbolt.org/z/rv4K37v3x

Reply:
Another alternative is of course just making a lambda. `std::bind` is a bit more concise, but less well known